server/src/main/kotlin/org/javacs/kt/externalsources/KlsURI.kt

```diff
@@ -93,11 +113,20 @@ data class KlsURI(val fileUri: URI, val query: Map<QueryParam, String>) {
         return result
     }
 
-    fun readContents(): String = withJarURLConnection {
-        it.jarFile
-            .getInputStream(it.jarEntry)
-            .bufferedReader()
-            .use(BufferedReader::readText)
+    fun readContents(): String {
+        return if (archiveType == ArchiveType.ZIP) {
+            val zipFile = ZipFile(File("$archivePath"))
+            zipFile.getInputStream(zipFile.getEntry(innerPath.trimStart('/')))
+                .bufferedReader()
+                .use(BufferedReader::readText)
+        } else {
+            withJarURLConnection {
+                it.jarFile
+                    .getInputStream(it.jarEntry)
+                    .bufferedReader()
+                    .use(BufferedReader::readText)
+            }
+        }
     }
```

Comment:
I would make this a `when` expression with an arm matching `ZIP` and another one matching `JAR, JDK` to make it explicit that we are dealing with a jar in the `else` branch (also to prevent oversights if we decide to add other archive types in the future), i.e. something along the lines of

```suggestion
    fun readContents(): String = when (archiveType) {
        ArchiveType.ZIP -> {
            val zipFile = ZipFile(File("$archivePath"))
            zipFile.getInputStream(zipFile.getEntry(innerPath.trimStart('/')))
                .bufferedReader()
                .use(BufferedReader::readText)
        }
        ArchiveType.JAR, ArchiveType.JDK -> {
            withJarURLConnection {
                it.jarFile
                    .getInputStream(it.jarEntry)
                    .bufferedReader()
                    .use(BufferedReader::readText)
            }
        }
    }
```

(off the top of my head, haven't checked it)